src/Provider/RemainingTotalProvider.php

```diff
@@ -60,9 +62,9 @@ public function getTotalLeftToRefund(int $id, RefundType $type): int
         return $unitTotal - $refundedTotal;
     }
 
-    private function getRefundUnitTotal(int $id, RefundType $refundType): int
+    private function getRefundUnitTotal(int $id, RefundTypeInterface $refundType): int
     {
-        if ($refundType->equals(RefundType::orderItemUnit())) {
+        if ($refundType instanceof Enum && $refundType->equals(RefundType::orderItemUnit())) {
```

Comment:
It would probably be worth introducing `equals` method from `Enum` class on `RefundTypeInterface`. This way, we would not have to check everywhere is `RefundType` an enum or use the class instead of the abstraction

Reply:
I believe we still can have this equals method on the interface? Or is it a problem, because it's a final function on the `Enum` class?

Reply:
Its final function and it is not possible to extend it on child class